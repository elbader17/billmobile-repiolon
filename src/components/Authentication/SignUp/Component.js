import React from 'react';
import { StyleSheet, View, Modal } from 'react-native';
import { Input, Button } from "react-native-elements";
import PasswordInputText from 'react-native-hide-show-password-input';
import Confirmation from '../Confirmation';

class SignUp extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
          email: '',
          password: '',
          name:'',
          confirmPassword: '',
          confirmationEmail:'',
          selectedIndex: 0,
          validateData: true,
          validEmail: true,
          validPass: true,
        };
      }

    handleSignUp = () => {
        const { name, email, password, confirmPassword } = this.state;
        const attributes = {
            email:email,
            name:name,
        };
        const { signUp } = this.props;
        signUp(password, email,attributes);
        this.setState({showConfirmationModal: true});
        
    }

    handleConfirmationCode = () => {
        const { confirmationEmail, confirmationCode } = this.state;
        const { confirmCode } = this.props;
        confirmCode(confirmationEmail,confirmationCode,{})
    }

    isEmail = (email) => {
        var re = /^[a-zA-Z0-9]+@[a-zA-Z0-9]+\.[A-Za-z]+$/;
        var bool = re.test(String(email).toLowerCase());
        this.state.validEmail = !bool;
        this.state.validateData = !bool || this.state.validPass;
        return bool;
    }

    isPasswordCoorrect = (password) => {
        var strongRegex = new RegExp("^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})");
        var bool = strongRegex.test(password);
        this.state.validPass = !bool;
        this.state.validateData = !bool || this.state.validEmail;
        return (bool);
    }


    setName = (value) => this.setState({ name: value})
    setEmail = (value) => this.setState({ email: value })
    setPassword = (value) => this.setState({ password: value })
    setConfirmPassword = (value) => this.setState({ confirmPassword: value })
    
    render() {
        return(
            <View style={{margin: 20}}>
                <Input
                    label="Nombre"
                    onChangeText={ this.setName }
                    placeholder="Nombre"
                />
                <Input
                    label="Email"
                    onChangeText={this.setEmail}
                    placeholder="usuario@email.com"
                    onRef={r => { this.state.email = r }}
                    value={this.state.email}
                    editable={!this.props.fetching}
                    valid={this.isEmail(this.state.email)}
                    returnKeyType='next'
                />
            
                <PasswordInputText
                    label="Password"
                    leftIcon={{ type: 'font-awesome', name: 'lock' }}
                    onChangeText={ this.setPassword }
                    placeholder="Aa@-1234"
                    valid={this.isPasswordCoorrect(this.state.password)}
                    secureTextEntry
                />
                
                <PasswordInputText
                    label="ConfirmPassword"
                    leftIcon={{ type: 'font-awesome', name: 'lock' }}
                    onChangeText={ this.setConfirmPassword }
                    placeholder="Aa@-1234"
                    secureTextEntry
                />
                <Button
                    title='Submit'
                    disabled={this.state.validateData}
                    onPress={ this.handleSignUp }
                />
            
                
                <Modal visible={this.props.showConfirmationModal} >
                    <View style={styles.container} >
                        <Confirmation/>
                    </View>
                </Modal>
        </View>
        )

    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
});

export default SignUp;