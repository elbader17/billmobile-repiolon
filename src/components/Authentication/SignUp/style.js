import { StyleSheet } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { COLORS } from '../../../constants/colors';
import { FONTS } from '../../../constants/fonts';

const styles = StyleSheet.create({
  container:{
    paddingVertical: 10,
    paddingHorizontal: 20,
  },
  containerInputs:{
    alignItems: 'center',
  },
  textBoxBtnHolder: {
    position: 'relative',
    alignSelf: 'stretch',
    justifyContent: 'center',
    marginTop: 5
  },
  textFormatPass: {
    fontFamily: FONTS.latoSemiBold,
    fontSize: FONTS.size11,
    color: COLORS.red,
    borderWidth: 0.5,
    borderRadius: 3,
    borderColor: COLORS.red,
    textAlign: 'center',
    marginBottom: 3,
    opacity: 0.9
  },
  textFormatPassValid: {
    fontFamily: FONTS.latoSemiBold,
    fontSize: FONTS.size11,
    color: 'green',
    borderWidth: 0.5,
    borderRadius: 3,
    borderColor: 'green',
    textAlign: 'center',
    marginBottom: 3,
    opacity: 0.9
  },
  textFormatEmail: {
    fontFamily: FONTS.latoSemiBold,
    fontSize: FONTS.size11,
    color: COLORS.red,
    borderWidth: 0.5,
    borderRadius: 3,
    borderColor: COLORS.red,
    textAlign: 'center',
    marginBottom: 3,
    opacity: 0.9
  },
  textFormatEmailValid: {
    fontFamily: FONTS.latoSemiBold,
    fontSize: FONTS.size11,
    color: 'green',
    borderWidth: 0.5,
    borderRadius: 3,
    borderColor: 'green',
    textAlign: 'center',
    marginBottom: 3,
    opacity: 0.9
  },
  textConfirmPass: {
    fontFamily: FONTS.latoSemiBold,
    fontSize: FONTS.size11,
    color: COLORS.red,
    borderWidth: 0.5,
    borderRadius: 3,
    borderColor: COLORS.red,
    textAlign: 'center',
    marginBottom: 3,
    opacity: 0.9
  },
  textConfirmPassValid: {
    fontFamily: FONTS.latoSemiBold,
    fontSize: FONTS.size11,
    color: 'green',
    borderWidth: 0.5,
    borderRadius: 3,
    borderColor: 'green',
    textAlign: 'center',
    marginBottom: 3,
    opacity: 0.9
  },
  textBox: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size18,
    alignSelf: 'stretch',
    height: 45,
    paddingRight: 45,
    paddingLeft: 8,
    borderWidth: 1,
    paddingVertical: 0,
    borderColor: COLORS.gray,
    borderRadius: 2
  },
  textBoxPass: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size18,
    alignSelf: 'stretch',
    height: 43,
    paddingRight: 45,
    paddingLeft: 8,
  },
  visibilityBtn: {
    position: 'absolute',
    right: 3,
    height: 40,
    width: 35,
    padding: 6,
    marginTop: 2,
  },
  inputPass: {
    flexDirection: 'row',
    borderColor: COLORS.gray, 
    borderWidth: 1, 
    borderRadius: 2
  },
  btnImage: {
    resizeMode: 'contain',
    height: '100%',
    width: '100%',
    opacity: 0.4
  },
  submit: {
    backgroundColor: COLORS.red,
    width: 266,
    height: hp('7%'),
    borderRadius: 2,
  },
  submitDisabled: {
    backgroundColor: COLORS.gray,
    width: 266,
    height: hp('7%'),
    borderRadius: 2,
  },
  textRegular11GrayDark: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size11,
    color: COLORS.grayDark,
  },
  textRegular14WhiteBold: {
    fontFamily: FONTS.latoSemiBold,
    fontSize: FONTS.size14,
    color: COLORS.white
  },
  textRed: {
    color: COLORS.red
  },
});

export default styles;