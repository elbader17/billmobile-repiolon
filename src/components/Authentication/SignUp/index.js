import { connect } from 'react-redux'
import SignUp from './Component';
import { signUp } from '../../../app/authentication/actions';
import { confirmCode } from '../../../app/authentication/actions';

const mapStateToProps = state => ({
  showConfirmationModal: state.authentication.showConfirmationModal,
});

function mapDispatchToProps(dispatch) {
  return {
    signUp: (password, name,attributes) => dispatch(signUp(password, name, attributes)),
  };
}

const component = connect(
  mapStateToProps,
  mapDispatchToProps,
)(SignUp);

export default component;
