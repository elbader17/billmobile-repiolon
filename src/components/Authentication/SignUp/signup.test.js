import React from 'react';
import { shallow } from 'enzyme';
import SignUp  from './Component';

jest.mock('../../../constants/metrics', () => ({ METRICS: { heightHeader: {} } }));

const promise = Promise.resolve(0);
const signUpMock = jest.fn(() => promise);

test('render correctly component SignUp', () => {
  const component = shallow(<SignUp />);
  expect(component).toMatchSnapshot();
});

test('call the handleSignUp function', () => {
  const component = shallow(<SignUp signUp={signUpMock} />);
  const button = component.findWhere(node => node.prop('testID') === 'submitSignUp');
  button.simulate('press');
  expect(signUpMock).toBeCalled();
});

/* FALLA - DATOS POR DEFAULT PARA NO COMPLETAR CONSTANTEMENTE (CAMBIAR)
test('initial state with the correct values', () => {
  const initialState = {
    email: '',
    password: '',
    name:'',
    confirmPassword: '',
    hidePassword: true,
    hideConfirmPassword: true,
    messagePassFormat: false,
    messageEmailFormat: false,
    messageMatchPass: false,
    loading: false
  };
  const component = shallow(<SignUp />);
  const state = component.instance().state;
  expect(state).toEqual(initialState);
});*/

describe('button CREAR CUENTA', () => {
  test('button initially disabled', () => {
    const component = shallow(<SignUp signUp={signUpMock} />);
    const button = component.findWhere(node => node.prop('testID') === 'submitSignUp');
    button.simulate('press');
    expect(button.props().disabled).toBe(true);
  });
  
  test('button enabled with the correct data', () => {
    const component = shallow(<SignUp signUp={signUpMock} />);
    component.setState({ 
      email: 'martin@gmail.com',
      password: '@Martin55',
      name: 'Martin',
      confirmPassword: '@Martin55'
    });
    const button = component.findWhere(node => node.prop('testID') === 'submitSignUp');
    expect(button.props().disabled).toBe(false);
  });
  
  test('button disabled with email incorrect', () => {
    const component = shallow(<SignUp signUp={signUpMock} />);
    component.setState({ 
      email: 'martingmail.com',
      password: '@Martin55',
      name: 'Martin',
      confirmPassword: '@Martin55'
    });
    const button = component.findWhere(node => node.prop('testID') === 'submitSignUp');
    expect(button.props().disabled).toBe(true);
  });
  
  test('button disabled with password incorrect', () => {
    const component = shallow(<SignUp signUp={signUpMock} />);
    component.setState({ 
      email: 'martin@gmail.com',
      password: '@Martin',
      name: 'Martin',
      confirmPassword: '@Martin55'
    });
    const button = component.findWhere(node => node.prop('testID') === 'submitSignUp');
    expect(button.props().disabled).toBe(true);
  });
  
  test('button disabled with confirmPassword incorrect', () => {
    const component = shallow(<SignUp signUp={signUpMock} />);
    component.setState({ 
      email: 'martin@gmail.com',
      password: '@Martin55',
      name: 'Martin',
      confirmPassword: '@Martin'
    });
    const button = component.findWhere(node => node.prop('testID') === 'submitSignUp');
    expect(button.props().disabled).toBe(true);
  });
  
  test('button disabled with password and confirmPassword no match', () => {
    const component = shallow(<SignUp signUp={signUpMock} />);
    component.setState({ 
      email: 'martin@gmail.com',
      password: '@Martin55',
      name: 'Martin',
      confirmPassword: '@Martin556'
    });
    const button = component.findWhere(node => node.prop('testID') === 'submitSignUp');
    expect(button.props().disabled).toBe(true);
  });
  
  test('button disabled with name empty', () => {
    const component = shallow(<SignUp signUp={signUpMock} />);
    component.setState({ 
      email: 'martingmail.com',
      password: '@Martin55',
      name: '',
      confirmPassword: '@Martin55'
    });
    const button = component.findWhere(node => node.prop('testID') === 'submitSignUp');
    expect(button.props().disabled).toBe(true);
  });
  
  test('button loading after press', () => {
    const component = shallow(<SignUp signUp={signUpMock} />);
    const button = component.findWhere(node => node.prop('testID') === 'submitSignUp');
    button.simulate('press');
    expect(component.instance().state.loading).toBe(true);
  });
  
});