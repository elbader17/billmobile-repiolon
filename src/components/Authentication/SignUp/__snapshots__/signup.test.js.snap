// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Component SigUp renders correctly SignUp 1`] = `ShallowWrapper {}`;
